### DISCUSSION TITLE
IfLetStore, Navigation to previous screen and logic error

### DISCUSSION THREAD

**Discussion by dmikots** [pointfreeco/swift-composable-architecture#1045] [Q&A] [answered]
Hi!
I need have some problem with optional state.
When I use navigation forward - everything ok. But, when I need to close 5 screen (deinit 5 state), a have a problem, that .binding() and .onAppear still work( but state is nil).
I try to use delay and Effect. 
![image](https://user-images.githubusercontent.com/57399131/161123441-c7e89808-7f67-4c45-b303-0ce9708c5553.png)
When delay between deinit state is 1 sec - I have 0 warnings.
But 
![telegram-cloud-photo-size-2-5345801869433354300-y](https://user-images.githubusercontent.com/57399131/161123799-887fd0ea-67d5-4680-8fd9-eb41401842d9.jpg)
now I have warnings.
How can I use option state, when I need to close and deinit Screen1 <- Screen2 <- Screen3 <- Screen 4 <- Screen5 ( action)

**Comment by stephencelis** [comment, marked as answer]
@dmikots There are a few cases where these runtime warnings are triggered by SwiftUI and are not an application logic error, and this is one of them.

In the future we hope to work around some of these issues, but for now you can safely ignore this warning.
